src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1784,7 +1784,10 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                 ImmutableArray<MethodSymbol> originalUserDefinedConversions = conversion.OriginalUserDefinedConversions;
                 if (originalUserDefinedConversions.Length > 1)
                 {
-                    diagnostics.Add(ErrorCode.ERR_AmbigUDConv, syntax.Location, originalUserDefinedConversions[0], originalUserDefinedConversions[1], operand.Type, targetType);
+                    // Method groups always have null types
+                    var operandType = (operand.Kind == BoundKind.MethodGroup) ? (object)MessageID.IDS_SK_METHOD.Localize() : operand.Type;
```

Comment:
>var operandType = (operand.Kind == BoundKind.MethodGroup) ? (object)MessageID.IDS_SK_METHOD.Localize() : operand.Type; [](start = 20, length = 118)

I think we should simply use operand.Display if operand.Type is null. I think this is the pattern we are following in the codebase, see GetAnonymousTypeFieldType function, for example.